joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/utils/FileUtil.scala

```diff
@@ -330,16 +268,25 @@ object FileUtil {
     }
   }
 
-  def using[R <: AutoCloseable, T](resource: => R)(f: R => T): T = {
-    val r = resource
+  /**
+   * Ensures that a resource is properly closed after use, even if an exception occurs.
+   * Similar to Java's try-with-resources but with more control.
+   *
+   * @param resource The resource to use
+   * @param f The function to apply to the resource
+   * @tparam R The resource type
+   * @tparam A The return type
+   * @return The result of applying f to the resource
+   */
+  def using[R <: AutoCloseable, A](resource: R)(f: R => A): A = {
```

Comment:
We can't simply use https://www.scala-lang.org/api/current/scala/util/Using$.html?

Reply:
I should have marked this PR as a draft again. I've been struggling with getting windows tests to pass due to files still being in use causing errors to happen on the console tests. Some of things in this PR is just trying to get the tests to pass on the runners, I will try removing this and using standard scala `Using` and see if it still passes all the tests.

Reply:
Reverted to using standard `Using`, seems to have still worked.